src/MockPublisher.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure;
+
+final class MockPublisher implements PublisherInterface
```

Comment:
```suggestion
final class CallbackPublisher implements PublisherInterface
```

So maybe should we switch to generic name again?

Reply:
I think we should keep "mock" so we don't encourage the use of it outside tests.

Reply:
And why cannot we use an anonymous class in API Platform tests?

Reply:
we can, and already doing so ( https://github.com/azjezz/core/blob/mercure-bundle-0.3/tests/Bridge/Doctrine/EventListener/PublishMercureUpdatesListenerTest.php#L543-L566 ), but this will leave us with less code + can be useful for others for testing purposes ( also the prefix `Mock` is consistent with the HttpClient component )